### PR TITLE
Add syntax highlighting for Vim

### PR DESCRIPTION
**Pull request by Vital7788**
and also Neovim

### CODE REVIEW

File: syntax/vim/syntax/assembly.vim [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+syn match asmInstruction	"cln"
+syn match asmInstruction	"clv"
+syn match asmInstruction	"strmv"
+syn match asmInstruction	"stregmv"
+syn match asmInstruction	"strmemv"
+syn match asmInstruction	"swap"
+
+syn match asmComment		";.*" contains=@Spell
+
+syn match asmDirective		"\.[A-Za-z][0-9A-Za-z-_]*"
```
Comment (Zefinder) [inline syntax/vim/syntax/assembly.vim:58]: Is it possible to add asm labels? A label is a directive but with a column at the end

**Review by Zefinder — COMMENTED** [review]
Thank you for the work!! But Vital, where are the unit tests?

File: syntax/vim/syntax/assembly.vim (hunk 1 above)
Comment (Vital7788) [reply syntax/vim/syntax/assembly.vim:58]: Do labels need to be preceded with a dot, like directives?

Comment (Zefinder) [reply syntax/vim/syntax/assembly.vim:58]: A label can begin with a point but it's not mandatory

Comment (Zefinder) [reply syntax/vim/syntax/assembly.vim:58]: Let's say that it doesn't so it's faster and I'm afraid of shift reduce conflicts

**Review by Zefinder — APPROVED** [review]

**Pull request merged**